### ISSUE TITLE
Subscribing to a Subject in a concatMap breaks RxJS

### ISSUE THREAD

**Issue by Sawtaytoes** [ReactiveX/rxjs#5410]
## Bug Report

**Current Behavior**
When subscribing to a `Subject` in a `concatMap`, it breaks Node.js when processing is complete even if the subject is still being listened. This might even extend past `concatMap` as I had the same problem in a different process where I created a subject, and Node.js died when it saw my code wasn't gonna `.next` it. It's not the expectation I'd have.

**Reproduction**
- REPL or Repo link:
https://github.com/Sawtaytoes/challenge-me/tree/7de12a0026f06bbf1b841bef0e54d5e06dc56825/packages/rxjs-challenges/scripts/tasks/diningPhilosophers

_Comment this line to reproduce the issue:
https://github.com/Sawtaytoes/challenge-me/blob/7de12a0026f06bbf1b841bef0e54d5e06dc56825/packages/rxjs-challenges/scripts/tasks/diningPhilosophers/createWantsToEatObservable.js#L220_

```js
concatMap(() => (
  someSubject$
))
```

**Expected behavior**
What should happen is that the Node.js process keeps running expecting _something_ could execute another `.next` on that subject. Or at least, `complete` or `error` the observable that made the subscription instead of killing the process.

**Environment**
- Runtime: Node `v10.19.0` and `v14.0.0`.
- RxJS version: `v6.5.5`

**Possible Solution**
The only solution I found was to unsubscribe from the subject after I was doing processing with it. Then the parent observable completed, `concatMap` worked fine, and Node.js didn't error.

```js
someSubject$
.pipe(
  // ... some stuff here that returns `EMPTY` or a value if everything works out.
  tap(() => {
    someSubject$
    .complete()
  }),
)
```

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

Expected:
![image](https://user-images.githubusercontent.com/3948069/80342432-ccd88500-8829-11ea-9a2b-36b49150a060.png)

Broken:
![image](https://user-images.githubusercontent.com/3948069/80342478-dcf06480-8829-11ea-9845-977eca25f2b3.png)

**Comment by cartant** [comment]
This issue needs a minimal reproduction.

**Comment by Sawtaytoes** [comment]
Here's a simpler example:

```js
const { concatMap, tap } = require('rxjs/operators')
const { merge, Subject, of } = require('rxjs')

const someSubject$ = new Subject()

const createSubjectListener = (
  value,
) => (
  merge(
    of(value),
    someSubject$,
  )
  .pipe(
    // Comment out this tap to break your app.
    tap(() => {
      someSubject$
      .complete()
    }),
  )
)

of(1, 2, 3)
.pipe(
  concatMap(
    createSubjectListener
  )
)
.subscribe({
  complete: () => {
    console.info('COMPLETE')
  },
  next: console.info,
})
```

The difference between `someSubject$.complete()` not being there and being there respectively:
![image](https://user-images.githubusercontent.com/3948069/80892439-aedaac80-8c8f-11ea-81a2-541ed51ab752.png)

**Comment by cartant** [comment]
This is not a bug. It's the expected behaviour. An incomplete observable is not going to prevent Node.js from terminating.

**Comment by Sawtaytoes** [comment]
Wait, what are you talking about? The bug is that Node.js _is_ terminating unexpectedly. It should _never_ terminate when there's an active listener.

Notice it says "clean exit" not `^C`.

**Comment by Sawtaytoes** [comment]
I think I get it now. You're saying because the observable is incomplete, Node.js notices there's nothing else being put on the stack and kills the running process?

**Comment by cartant** [comment]
Node notices nothing. Observables are just functions that are chained together. They are not event sources - like `setInterval`, etc. - as far as Node is concerned. Unless an observable wraps an actual Node event source, Node knowns nothing about it. Subscriptions are not event listeners.

**Comment by Sawtaytoes** [comment]
Something's really weird here though. If it's still processing; as in, I called `.next`, so it shouldn't kill the the entire Node.js process right? Because it should still be processing and going through the observable pipeline.

I tried something today in a different codebase where I swapped the positions where `subject$`  was first and `of` second in `merge`. I got the data I was looking for but only got one output with `of` first and `subject$` second.

I tried it with that code example, got `1` both times, but that's because my original code to test this had a `subject$.next()` in it.

I commented out the `.complete` and put in a `.next(value + 1)` and got nothing logged (correct, because it's an infinite loop). I would've expected that same behavior if the order was swapped.

What's going on here?
